diff --git a/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts b/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts
index ea2794b..3b42921 100644
--- a/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts
@@ -29,6 +29,8 @@
 			pinctrl-names = "default";
 			pinctrl-0 = <&spi1_pins &spi1_cs_pins>;
 			cs-gpios = <&gpio 18 1>;
+			interrupts = <1 29>;
+			interrupt-parent = <&intc>;
 			status = "okay";
 
 			spidev1_0: spidev@0 {
diff --git a/arch/arm/boot/dts/overlays/uart1-overlay.dts b/arch/arm/boot/dts/overlays/uart1-overlay.dts
index 986d725..9628995 100644
--- a/arch/arm/boot/dts/overlays/uart1-overlay.dts
+++ b/arch/arm/boot/dts/overlays/uart1-overlay.dts
@@ -9,6 +9,9 @@
 		__overlay__ {
 			pinctrl-names = "default";
 			pinctrl-0 = <&uart1_pins>;
+			interrupts = <1 29>;
+			interrupt-parent = <&intc>;
+			autoload = "okay";
 			status = "okay";
 		};
 	};
