diff --git a/boot/pxe_utils.c b/boot/pxe_utils.c
index a7a84f26c1..cc8e62325e 100644
--- a/boot/pxe_utils.c
+++ b/boot/pxe_utils.c
@@ -500,8 +500,10 @@ static int label_boot(struct pxe_context *ctx, struct pxe_label *label)
 		char bootargs[CONFIG_SYS_CBSIZE] = "";
 		char finalbootargs[CONFIG_SYS_CBSIZE];
 
+		/* NOTE: We do not want to overwrite the basebootargs, instead append them, */
+		/*       so we are able to do both to keep and to overwrite.                */
 		if (strlen(label->append ?: "") +
-		    strlen(ip_str) + strlen(mac_str) + 1 > sizeof(bootargs)) {
+		    strlen(ip_str) + strlen(mac_str) + strlen(env_get("basebootargs")?:"") + 1 > sizeof(bootargs)) {
 			printf("bootarg overflow %zd+%zd+%zd+1 > %zd\n",
 			       strlen(label->append ?: ""),
 			       strlen(ip_str), strlen(mac_str),
@@ -509,8 +511,9 @@ static int label_boot(struct pxe_context *ctx, struct pxe_label *label)
 			return 1;
 		}
 
-		if (label->append)
-			strncpy(bootargs, label->append, sizeof(bootargs));
+		if (label->append) {
+			sprintf(bootargs, "%s %s", env_get("basebootargs")?:"", label->append);
+		}
 
 		strcat(bootargs, ip_str);
 		strcat(bootargs, mac_str);
@@ -649,6 +652,15 @@ static int label_boot(struct pxe_context *ctx, struct pxe_label *label)
 		zboot_argc = 5;
 	}
 
+	if (!bootm_argv[2]) {
+		/* if no initrd label is passed assume it's our common fitimage where initrd is part of it so pass just the address */
+		const char* ramdisk_addr = env_get("ramdisk_addr_r");
+		if (ramdisk_addr && strcmp("", ramdisk_addr)) {
+			bootm_argv[2] = ramdisk_addr;
+			bootm_argc = 3;
+		}
+	}
+
 	if (!bootm_argv[3])
 		bootm_argv[3] = env_get("fdt_addr");
 
